db/fixtures/ae_datastore/ManageIQ/Cloud/VM/Retirement/StateMachines/Methods.class/__methods__/amazon_pre_retirement.rb

```diff
@@ -4,12 +4,26 @@
 #
 
 vm = $evm.root['vm']
-if vm && vm.attributes['power_state'] == 'on'
-  ems = vm.ext_management_system
+ems = vm.ext_management_system if vm
+
+if vm.nil? || ems.nil?
+  $evm.log('info', "Skipping Amazon pre retirement for Instance:<#{vm.try(:name)}> on EMS:<#{ems.try(:name)}>")
+  exit MIQ_OK
+end
+
+power_state = vm.power_state
+if power_state == "on"
+  if vm.hardware.root_device_type.blank?
+    $evm.log('error', "Aborting Amazon pre retirement for <#{vm.name}> with type:<#{vm.hardware.root_device_type}>")
+    $evm.log('error', "Aborting Amazon pre retirement, Instance may have been provisioned outside of CFME <#{vm.name}>")
```

Comment:
@billfitzgerald0120 
can we combine these 2 logs into 1, the first log line is going to have nil for the device_type
$evm.log('error', "Aborting Amazon pre retirement, empty root_device_type. Instance may have been provisioned outside of CFME <#{vm.name}>"